client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -176,6 +176,95 @@ export const PatternLibrary = ( {
 
 	const isHomePage = ! category && ! searchTerm;
 
+	const PATTERN_SEO_CONTENT: Record< Category[ 'name' ], { title: string } > = {
+		default: {
+			title: translate( 'WordPress Patterns', {
+				comment: 'Pattern Library home page title',
+				textOnly: true,
+			} ),
+		},
+		header: {
+			title: translate( 'WordPress Header Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		footer: {
+			title: translate( 'WordPress Footer Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		about: {
+			title: translate( 'WordPress About Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		posts: {
+			title: translate( 'WordPress Blog Post Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		contact: {
+			title: translate( 'WordPress Contact Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		events: {
+			title: translate( 'WordPress Events Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		gallery: {
+			title: translate( 'WordPress Gallery Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		intro: {
+			title: translate( 'WordPress Intro Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		menu: {
+			title: translate( 'WordPress Menu Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		newsletter: {
+			title: translate( 'WordPress Newsletter Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		services: {
+			title: translate( 'WordPress Services Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		store: {
+			title: translate( 'WordPress Store Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+		testimonials: {
+			title: translate( 'WordPress Testimonial Patterns', {
+				comment: 'Pattern Library category page title',
+				textOnly: true,
+			} ),
+		},
+	};
```

Comment:
Either that, or we move this, and `<DocumentHead title={ seoContent.title } />` into its own component.

Moving this array outside the function body is slightly complex because we need to pass the `translate` function for that to work (we could theoretically import `translate` directly, but `useTranslate` is the recommended approach). #88997 could serve as inspiration for how to do that, though.

Reply:
Holding off on changes here, as I know there's still ongoing disucssion over in #88997, but I'm happy to pursue whichever course y'all think best!

Reply:
Let's go with adding a new component that wraps `DocumentHead`. Within that component, we can define `PATTERN_SEO_CONTENT` in the function scope, no problem